src/HttpClient.php

```diff
@@ -91,6 +93,67 @@ public function deploy(array $data)
         $this->guzzle->post('deploy', ['json' => $data]);
     }
 
+    /**
+     * Notify Bugsnag of a build.
+     *
+     * @param array $data the build information
+     *
+     * @return void
+     */
+    public function sendBuildReport(array $buildInfo)
+    {
+        $app = $this->config->getAppData();
+
+        $data = [];
+        $sourceControl = [];
+        if (isset($buildInfo['repository'])) {
+            $sourceControl['repository'] = $buildInfo['repository'];
+        }
+
+        if (isset($buildInfo['provider'])) {
+            $sourceControl['provider'] = $buildInfo['provider'];
+        } elseif (isset($buildInfo['repository'])) {
+            $url = $buildInfo['repository'];
+            if (strpos($url, 'github.com') != false) {
+                $sourceControl['provider'] = 'github';
+            } elseif (strpos($url, 'bitbucket.com') != false) {
+                $sourceControl['provider'] = 'bitbucket';
+            } elseif (strpos($url, 'gitlab.com')) {
+                $sourceControl['provider'] = 'gitlab';
+            }
+        }
+
+        if (isset($buildInfo['revision'])) {
+            $sourceControl['revision'] = $buildInfo['revision'];
+        }
+
+        if (!empty($sourceControl)) {
+            $data['sourceControl'] = $sourceControl;
+        }
+
+        if (isset($buildInfo['builder'])) {
+            $data['builderName'] = $buildInfo['builder'];
```

Comment:
This should default to `whoami` or a php cross platform equivalent.

Reply:
I don't think we want to add shell calls in the base notifier, it's much safer to use wrapped commands in higher level libs.

Reply:
Why not? If this isnt automated very few people will set it.

Reply:
You probably want:

```php
get_current_user();
```

instead of shell executing `whoami`. That function will give you the owner of the file, which is maybe better than nothing.

Something you could do if the user has installed and not disabled the functions from posix extension, is:

```php
posix_getpwuid(posix_geteuid())['name'];
```

Reply:
`whoami` works across all windows post server-2003 & unix-like systems.  I'll add a check on the response to see if it returned correctly and fallback to `get_current_user` if not.